src/python/pants/engine/internals/rule_visitor.py

```diff
@@ -244,14 +244,70 @@ def _get_awaitable(self, call_node: ast.Call, is_effect: bool) -> AwaitableConst
             is_effect,
         )
 
+    def _get_byname_awaitable(
+        self, rule_func: Callable, call_node: ast.Call
+    ) -> AwaitableConstraints:
+        parse_error = partial(
+            GetParseError, get_args=call_node.args, source_file_name=self.source_file
+        )
+
+        output_type = _lookup_return_type(rule_func, check=True)
+
+        if call_node.args:
+            raise parse_error(
+                self._format(
+                    call_node,
+                    (
+                        f"Expected no positional arguments (got {len(call_node.args)}), and "
+                        "a `**implicitly(..)` application."
+                    ),
+                )
+            )
+
+        input_types: tuple[type, ...]
+        if not call_node.keywords:
+            input_types = ()
+        elif (
+            len(call_node.keywords) == 1
+            and not call_node.keywords[0].arg
+            and isinstance(implicitly_call := call_node.keywords[0].value, ast.Call)
```

Comment:
Am I reading correctly that nowhere do we verify that this is in fact a call to `implicitly(...)` and it could be a call to anything with kwargs?

Reply:
Mm, yea: good eye. This code was written before the `implicitly(..)` keyword was even introduced. I'll add a constraint.